script/postinstall.cmd

```diff
@@ -13,3 +13,5 @@ echo ^>^> Rebuilding apm dependencies with bundled Node !bundledVersion!
 setx JOBS 16
 
 call .\bin\npm.cmd rebuild
+
+exit /b
```

Comment:
So, this one honestly still confuses me the most.

<details><summary>Hidden behind an expander, because here lies madness (click to expand if you dare):</summary>

_(The other two I can sigh about and move on, it's just calling `.cmd` scripts from other `.cmd` scripts being super clunky and easily breaking things if not done just so. But at least the solution is at least kinda well documented... Some parts of it are a mystery to me, but only niche aspects of it, and I get the sense it's getting deep enough into it and there lies only madness and undefined/undocumented behavior beyond a certain point.)_

About `exit /b` ... the `/b` flag is only _supposed_ to mean "exit the current subroutine, not the whole `cmd.exe`." That much I can just about accept. (Why is this not the default though??? Why is exiting the entire `cmd.exe` the same lever as exiting a subroutine, and making the distinction is opt-in with the `/b` flag? Seems super error-prone... /grumpiness).

Here's the weird part, we're using it for something that I'm not sure it's supposed to do (hiding/resetting a previous error code).

It lets you explicitly exit with a certain error code, such as `exit /b 6` for error code 6, a number I just made up. _`exit /b` without an explicit code is supposed to carry forward the previous error code, if you don't explicitly set one like `exit /b [code]`._

Sources:
- https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/exit
  - Official docs, doesn't really clarify what happens to a previous/existing error code if you don't specify a new one.
  - (The official docs are way too short, IMO, they leave a lot of these finer questions un-addressed.)
- https://ss64.com/nt/exit.html
  - Some madman ([Simon Sheppard](https://ss64.com/docs/), apparently), who I have to trust implicitly, since manual testing seems to be the main way to verify _anything_ about `cmd.exe`, and this seems (???) to be the product of much manual testing and mad note-taking and scribbling.
    - > `EXIT` without an ExitCode acts the same as [`goto:eof`](https://ss64.com/nt/goto.html) and will not alter the [ERRORLEVEL](https://ss64.com/nt/errorlevel.html)
      - Also this (https://ss64.com/nt/goto.html), oh no:
      - > If the jump is successfully made <code><a href="https://ss64.com/nt/errorlevel.html">%ERRORLEVEL%</a> = <i>unchanged</i></code>, typically this will be 0 but if a previous command set an errorlevel, that will be preserved (this is a bug).
      - > **_(this is a bug)_** 🤦

And yet here I am doing `exit /b` here to suppress the error level from the `npm.cmd` --> `node_modules\.bin\npm.cmd` we are calling just above.

As far as I can read in any docs, this should do nothing. And yet I have tested it works _with_ this change and fails _without_ this change over and over. Due to the error level being visible to Yarn, and Yarn exiting non-zero, which would kill a CI run.

(Other than in CI, the error messages from the `npm rebuild` would just be a bit of a nuisance, since the C/C++ code seems to have built just fine??? I dunno how to verify all that built C/C++ code, honestly, but all tests are passing, right??? But failed CI runs is a bit of an issue, so I still think "hiding" the errors is better if they aren't actual problems. Like, maybe the `makefile` or equivalent for these C/C++ addons tries some commands that might be missing and does fallback behavior that works instead, but the original "command not found" error status persists long enough to trip up Yarn without this change??? I have to guess so hard at this, since it's increasingly deep down the rabbit hole just to fix a package manager issue.)

</details>

This, and having to scratch my head a lot to realize about the Yarn shims, were a large part of my descent into madness on this one (on this PR). I'm not even sure how/why npm avoids all these issues. And at this point, _I don't want to know_.

Reply:
Reading this really just makes me say, lets make all scripts JavaScript ones, with `#/env/node` at the top, and require everyone to have node already. That's just a lot of ridiculousness

Reply:
I have loosely aspired to replacing these shell and batch scripts with pure JS for some time. I assume it would've been harder ~9 years ago when apm was first written, but NodeJS has kind of had everything and the kitchen sink built in by now. Indeed `child_process.spawn()` and the like can set env vars and call out to CLI commands, generally replacing the main point of shell/batch scripts.

If we can do JS with the same or better reliability for ppm in-use, then it'll be worth it. Otherwise, if the shell/batch scripts run more reliably, then perhaps the insanity these scripts bring during maintenance can be a maintainer-only concern, and a rare one (just don't touch them that often)... Not that I love the thought of that. We'll see...

_(More cmd weirdness: You might be intrigued to hear that adding `exit /b` at the bottom of our `bin\npm.cmd`, closer to where the errors I was trying to suppress came from, seemed to do nothing. Whereas doing `exit /b` one further layer out, at the bottom of `postinstall.cmd`, did the trick. 🤷.)_